Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -333,6 +319,17 @@ static MTLVertexStepFunction stepFunction(WGPUVertexStepMode stepMode)
     return vertexDescriptor;
 }
 
+static auto buildKeyValueReplacements(const auto& stage)
+{
+    HashMap<String, float> keyValueReplacements;
```

Comment:
WebGPU.h indicates this should be a double, not a float

Reply:
It is true it is double in WGPU but Metal does not support double values and it is not clear from the definition of `-[MTLFunctionConstantValues setConstantValue:type:withName:];` which takes the address of the value, do I need to persist the value in memory until the shader is fully compiled or just for the duration of the `setConstantValue` method (in which case I could pass the address of a temporary)